app/build.gradle

```diff
@@ -466,3 +468,34 @@ if (project.hasProperty("coverage")) {
 afterEvaluate {
     check.dependsOn 'findbugs', 'pmd', 'checkstyle'
 }
+
+// -------------------------------------------------------------------------------------------------
+// Bitbar Cloud Testing: testDroidUpload
+// Upload both the application and test APK and conduct a UI test-run
+// Utilizes a preset group of Amazon Fire TV devices on a preset project and framework
+// -------------------------------------------------------------------------------------------------
+
+testdroid {
+
+    try {
+        def tokenFile = new File("${rootDir}/.bitbar_token.json")
+        def parsedFile = new groovy.json.JsonSlurper().parseText(tokenFile.text)
+
+        apiKey parsedFile.secret.bitbarToken.api_key
+        cloudUrl parsedFile.secret.bitbarToken.cloud_url
+
+        println "(Added from .bitbar_token file)"
+    } catch (FileNotFoundException ignored) {
+        println("X_X")
+    }
+
+    authorization "APIKEY"
+    deviceGroup "Amazon"
+    projectName "Firefox for Fire TV"
+    testRunName "Build: " + android.defaultConfig.versionName + " - " + new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date())
+    frameworkId 24
+    fullRunConfig {
+        instrumentationRunner = android.defaultConfig.testInstrumentationRunner
```

Comment:
Some of our tests will fail if run without using `org.mozilla.tv.firefox.FirefoxTestRunner`.

Reply:
I thought that is what is defined by defaultConfig in line 33 of build.gradle ?
`        testInstrumentationRunner "org.mozilla.tv.firefox.FirefoxTestRunner"`

If not, I can change that.

Reply:
It is, but what does this line do?  It looks like it's redeclaring the same information.  We saw earlier that BitBar and Firebase were ignoring some of what we declared in Gradle, so it's not implausible.

Reply:
The line just specifies a custom instrumentation runner framework (Default: `android.test.InstrumentationTestRunner`. It doesn't seem to be required, but I didn't see any problems leaving it in. If the default is fine, I can remove it?

http://docs.bitbar.com/testing/api/tests/index.html?highlight=instrumentationrunner

Reply:
I'm not honestly clear on what it's doing, but if everything passes without it that suggests to me that we should delete it.

Reply:
Does the test run on bitbar without 498? then `android.test.InstrumentationTestRunner` as default value is working, and we can remove line 498.  If not, then we should keep it.  Sometimes the cloud test frameworks needs to re-declare ones that are already defined in gradle, because they are disregarding most of it.

Reply:
If `instrumentationRunner` is empty the primary Android test runner class `android.test.InstrumentationTestRunner` is used. Yes, I can remove it.